build/bump.php

```diff
@@ -39,271 +32,257 @@ function usage($command)
-$dev_status = 'Stable';
-
-if (!isset($versionParts[1]))
-{
-	$versionParts[1] = '';
-}
-else
-{
-	if (preg_match('#^dev#', $versionParts[1]))
-	{
-		$dev_status = 'Development';
-	}
-	elseif (preg_match('#^alpha#', $versionParts[1]))
-	{
-		$dev_status = 'Alpha';
-	}
-	elseif (preg_match('#^beta#', $versionParts[1]))
-	{
-		$dev_status = 'Beta';
-	}
-	elseif (preg_match('#^rc#', $versionParts[1]))
-	{
-		$dev_status = 'Release Candidate';
+/**
+ * @param $rootPath
+ * @param $file
+ * @param $version
+ * @return void
+ */
+function updateLanguagePackXmlFile($rootPath, $file, $version) {
+	$filePath = $rootPath . $file;
+	if (file_exists($filePath)) {
+		$contents = file_get_contents($filePath);
+		$contents = preg_replace('#<version>[^<]*</version>#', '<version>' . $version['full'] . '</version>', $contents);
+		$contents = preg_replace('#<creationDate>[^<]*</creationDate>#', '<creationDate>' . $version['credate'] . '</creationDate>', $contents);
+		$contents = preg_replace(
+			'#<h2>(.*)<\/h2>#',
+			'<h2>Deutsches Sprachpaket (Version: ' . $version['full'] . ' vom ' . $version['credate_de'] . ') für Joomla! ' . $version['main'] . ' von <a title="J!German" href="https://www.jgerman.de" target="_blank" rel="noopener noreferrer">J!German</a></h2>',
+			$contents
+		);
+		file_put_contents($filePath, $contents);
+		echo "Updated: $filePath" . PHP_EOL;
+	} else {
+		echo "Skipped: File not found - $filePath" . PHP_EOL;
 	}
 }
 
-if (!isset($versionParts[2]))
-{
-	$versionParts[2] = '';
-}
-else
-{
-	$dev_status = 'Development';
+/**
+ * @param $rootPath
+ * @param $file
+ * @param $version
+ * @return void
+ */
+function updateLanguagePackSqlFile($rootPath, $file, $version) {
+	$filePath = $rootPath . $file;
+	if (file_exists($filePath)) {
+		$contents = file_get_contents($filePath);
+		$contents = preg_replace('#"version":"[^"]*"#', '"version":"' . $version['full'] . '"', $contents);
+		$contents = preg_replace('#"creationDate":"[^"]*"#', '"creationDate":"' . $version['credate'] . '"', $contents);
+		file_put_contents($filePath, $contents);
+		echo "Updated: $filePath" . PHP_EOL;
+	} else {
+		echo "Skipped: File not found - $filePath" . PHP_EOL;
+	}
 }
 
-// Set version properties.
-$versionSubParts = explode('.', $versionParts[0]);
+/**
+ * @param $rootPath
+ * @param $year
+ * @param $excludeDirectories
+ * @param $excludeFiles
+ * @return void
+ */
+function changeCopyrightDate($rootPath, $year, $excludeDirectories, $excludeFiles) {
+	$changedFiles = 0;
+	$directory = new RecursiveDirectoryIterator($rootPath);
+	$iterator = new RecursiveIteratorIterator($directory, RecursiveIteratorIterator::SELF_FIRST);
+
+	foreach ($iterator as $file) {
+		if ($file->isFile()) {
+			$filePath = $file->getPathname();
+			$relativePath = str_replace($rootPath, '', $filePath);
+
+			// Skip excluded files and directories
+			if (preg_match('#\.(png|jpeg|jpg|gif|bmp|ico|webp|svg|woff|woff2|ttf|eot)$#', $filePath) || in_array($relativePath, $excludeFiles)) {
```

Comment:
I do get the idea here but wouldn't it be easier to list only the extensions to include rather than the extensions to exclude?

Reply:
This was taken from the original, but of course it should be adapted!